packages/eslint-plugin-svelte/src/rules/consistent-selector-style.ts

```diff
@@ -0,0 +1,292 @@
+		if (styleValue === 'id' && couldBeId(selection)) {
+			ruleGlobals.context.report({
+				messageId: 'classShouldBeId',
+				loc: ruleGlobals.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'type' && couldBeType(selection, ruleGlobals.typeSelections)) {
+			ruleGlobals.context.report({
+				messageId: 'classShouldBeType',
+				loc: ruleGlobals.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+	}
+}
+
+/**
+ * Checks an ID selector
+ */
+function checkIdSelector(node: SelectorIdentifier, ruleGlobals: RuleGlobals): void {
+	const selection = ruleGlobals.idSelections.get(node.value) ?? [];
+	for (const styleValue of ruleGlobals.style) {
+		if (styleValue === 'class') {
+			ruleGlobals.context.report({
+				messageId: 'idShouldBeClass',
+				loc: ruleGlobals.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'id') {
+			return;
+		}
+		if (styleValue === 'type' && couldBeType(selection, ruleGlobals.typeSelections)) {
+			ruleGlobals.context.report({
+				messageId: 'idShouldBeType',
+				loc: ruleGlobals.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+	}
+}
+
+/**
+ * Checks a type selector
+ */
+function checkTypeSelector(node: SelectorTag, ruleGlobals: RuleGlobals): void {
+	const selection = ruleGlobals.typeSelections.get(node.value) ?? [];
+	for (const styleValue of ruleGlobals.style) {
+		if (styleValue === 'class') {
+			ruleGlobals.context.report({
+				messageId: 'typeShouldBeClass',
+				loc: ruleGlobals.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'id' && couldBeId(selection)) {
+			ruleGlobals.context.report({
+				messageId: 'typeShouldBeId',
+				loc: ruleGlobals.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'type') {
+			return;
+		}
+	}
+}
+
+/**
+ * Checks whether a given selection could be obtained using an ID selector
+ */
+function couldBeId(selection: AST.SvelteHTMLElement[]): boolean {
+	return selection.length <= 1;
+}
+
+/**
+ * Checks whether a given selection could be obtained using a type selector
+ */
+function couldBeType(
+	selection: AST.SvelteHTMLElement[],
+	typeSelections: Map<string, AST.SvelteHTMLElement[]>
+): boolean {
+	const types = new Set(selection.map((node) => node.name.name));
+	if (types.size > 1) {
+		return false;
+	}
+	if (types.size < 1) {
+		return true;
+	}
+	const type = [...types][0];
+	const typeSelection = typeSelections.get(type);
+	return typeSelection !== undefined && arrayEquals(typeSelection, selection);
+}
+
+/**
+ * Compares two arrays for item equality
+ */
+function arrayEquals(array1: AST.SvelteHTMLElement[], array2: AST.SvelteHTMLElement[]): boolean {
+	function comparator(a: AST.SvelteHTMLElement, b: AST.SvelteHTMLElement): number {
+		return a.range[0] - b.range[0];
+	}
+
+	const array2Sorted = array2.slice().sort(comparator);
+	return (
```

Comment:
Couldn't you compare it like this?

```js
return (
  array1.length === array2.length && 
  array1.every(e => array2.includes(e))
)
```

Reply:
You could, but it's slower (n² instead of n log n)

Reply:
In my opinion, I don't expect any performance issues with non-malicious `.svelte` files (please let me know if this isn't the case).
For now I prefer to keep the function size at 2 lines.